src/storage/txn/process.rs

```diff
@@ -523,13 +523,17 @@ fn process_write_impl<S: Snapshot, L: LockMgr>(
             let mut locks = vec![];
             let rows = mutations.len();
 
+            // We should not call `prewrite_key_value` anymore if any key has been locked
+            let mut skip_kv = false;
+
             // If `options.for_update_ts` is 0, the transaction is optimistic
             // or else pessimistic.
             if options.for_update_ts == 0 {
                 for m in mutations {
-                    match txn.prewrite(m, &primary, &options) {
+                    match txn.prewrite(m, &primary, &options, skip_kv) {
                         Ok(_) => {}
                         e @ Err(MvccError::KeyIsLocked { .. }) => {
+                            skip_kv = true;
                             locks.push(e.map_err(Error::from).map_err(StorageError::from));
                         }
                         Err(e) => return Err(Error::from(e)),
```

Comment:
This interface seems to be ugly. I think we could split function `prewrite`. 
```
for m in mutations {
      if txn.check_constraint_and_lock(&m, &primary, &options) {
           locks.push(e.map_err(Error::from).map_err(StorageError::from));
      }
}
if locks.empty() {
    for m in mutations {
        txn.prewrite_key_value(m, &options);
   }
}
```

Reply:
How about to change the prewrite method to `fn prewrite(mutations: &[..], ...) -> ...`

Reply:
I think it's still better than use `flag` to skip. But all interface of `MvccTxn` is about **one single key**, rather than dealing with multiple key together.